libsignal-protocol/src/lib.rs

```diff
@@ -133,3 +128,11 @@ pub trait Serializable {
         Ok(())
     }
 }
+
+/// A helper trait for something which can be deserialized from protobufs.
+pub trait Deserializable {
```

Comment:
Instead of having a `where Self: Sized` bound on `deserialize()` you should declare `Deserializable` as requiring `Sized` (i.e. `trait Deserializable: Sized { ... }`). 

That would line up with [`serde::Deserialize`](https://docs.rs/serde/1.0.104/serde/trait.Deserialize.html). Plus where clauses on trait methods are usually used to let the method opt-out of being object-safe, which doesn't really make sense considering it's got one function which will return an instance of `Self` on the stack.